src/main/java/uk/ac/ebi/ega/egafuse/service/EgaDirectory.java

```diff
@@ -83,7 +85,12 @@ public synchronized void read(Pointer buf, FuseFillDir filler) {
             if (getName().equalsIgnoreCase("datasets")) {
```

Comment:
This is generally a risky way to use the equals methods, I think we should take the chance to make it NPE safe